core-relations/src/free_join/mod.rs

```diff
@@ -529,21 +544,39 @@ impl Database {
         ever_changed
     }
 
+    /// A "fast path" merge method that is not optimized for parallelism and does not respect read
+    /// and write dependencies. This ends up being faster than the full "strata-aware" option in
+    /// the body of `merge_all`.
+    fn merge_simple(&mut self, mut to_merge: SmallVec<[TableId; 4]>) -> bool {
+        let mut changed = false;
+        while !to_merge.is_empty() {
+            for table_id in to_merge.iter().copied() {
```

Comment:
A small number of tables to merge does not imply a small or a balanced workload, e.g., the AC workload, so parallelization can still help. Should we pass the parallelization flag from the call site and optionally do `par_iter` here?

Reply:
on the python-array-optimize test, this command:

```
time target/release/egglog tests/python_array_optimize.egg > /tmp/x
```
Goes from 1.2s with merge_simple commented out to 0.9s with the merge_simple path enabled (on my m4 max laptop)

I'll change it to 4 though!